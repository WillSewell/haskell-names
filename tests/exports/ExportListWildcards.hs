module ExportListWildcards (Foo(..), Bar(..), N(..)) where

data Foo = Foo1 | Foo2 Int | Foo3 { c :: Bool }

class Bar a where x :: a -> Foo

newtype N = N { unN :: Foo }

a = a

data Baz = Baz
