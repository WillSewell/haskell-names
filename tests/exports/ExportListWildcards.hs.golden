Symbols
  (fromList
     [ SymMethod
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "x" }
               }
         , sv_fixity = Nothing
         , sv_className =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Bar" }
               }
         }
     , SymSelector
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "c" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo" }
               }
         , sv_constructors =
             [ OrigName
                 { origPackage = Nothing
                 , origGName =
                     GName { gModule = "ExportListWildcards" , gName = "Foo3" }
                 }
             ]
         }
     , SymSelector
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "unN" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "N" }
               }
         , sv_constructors =
             [ OrigName
                 { origPackage = Nothing
                 , origGName =
                     GName { gModule = "ExportListWildcards" , gName = "N" }
                 }
             ]
         }
     , SymConstructor
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo1" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo" }
               }
         }
     , SymConstructor
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo2" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo" }
               }
         }
     , SymConstructor
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo3" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo" }
               }
         }
     , SymConstructor
         { sv_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "N" }
               }
         , sv_fixity = Nothing
         , sv_typeName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "N" }
               }
         }
     ])
  (fromList
     [ SymData
         { st_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Foo" }
               }
         , st_fixity = Nothing
         }
     , SymNewType
         { st_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "N" }
               }
         , st_fixity = Nothing
         }
     , SymClass
         { st_origName =
             OrigName
               { origPackage = Nothing
               , origGName =
                   GName { gModule = "ExportListWildcards" , gName = "Bar" }
               }
         , st_fixity = Nothing
         }
     ])